src/app/drive/services/file.service/uploadFile.ts

```diff
@@ -72,7 +73,21 @@ export async function uploadFile(
       encrypt_version: StorageTypes.EncryptionVersion.Aes03,
     };
 
-    const response = await storageClient.createFileEntry(fileEntry);
+    let response = await storageClient.createFileEntry(fileEntry);
+    if (!response.thumbnails) {
+      response = {
+        ...response,
+        thumbnails: []
+      };
+    }
+
+    const thumbnail = await generateThumbnailFromFile(file, response.id, userEmail, isTeam);
```

Comment:
Perhaps naming this as `generatedThumbnail` makes it clearer to then use `generatedThumbnail.thumbnail` instead of `thumbnail.thumbnail`